src/test/resources/resyncContext.xml

```diff
@@ -7,17 +7,18 @@
 
     <context:property-placeholder location="classpath:broker.properties"/>
 
-    <bean id="resyncService" class="org.esbtools.gateway.resync.ResyncService" >
-        <property name="resyncConfigurations">
-            <map>
-                <entry key="GitHub"><ref bean="gitHubSystemConfiguration"/></entry>
-            </map>
-        </property>
+    <bean id="gitHubResyncService" class="org.esbtools.gateway.resync.JmsResyncService">
+        <constructor-arg index="0" ref="gitHubSystemConfiguration" />
+    </bean>
+
+    <bean id="badHubResyncService" class="org.esbtools.gateway.resync.JmsResyncService">
+        <constructor-arg index="0" ref="badHubSystemConfiguration" />
     </bean>
 
     <bean id="gitHubSystemConfiguration" class="org.esbtools.gateway.resync.ResyncConfiguration">
-        <property name="endSystem" value="GitHub" />
-        <property name="broker" ref="amqJmsTemplate" />
+        <constructor-arg index="0" value="GitHub" />
+        <constructor-arg index="1" ref="amqJmsTemplate" />
+        <constructor-arg index="2" value="resync" />
     </bean>
```

Comment:
IMHO it is an extra layer to have the configuration be a separate object, but I also recognize it's not a huge deal, so if you prefer the config object that's fine with me personally.